app/controllers/collections_controller.rb

```diff
@@ -2,6 +2,8 @@
 
 # Controller for Collections
 class CollectionsController < ApplicationController
+  before_action :wasapi_provider_accounts
```

Comment:
I think this only needs to happen for :new and :edit, right?

Reply:
Yes. Fixed.